ziti/src/main/kotlin/org/openziti/net/nio/SyncSocketChannel.kt

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2018-2021 NetFoundry, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openziti.net.nio
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.runBlocking
+import org.openziti.Ziti
+import org.openziti.impl.ZitiContextImpl
+import org.openziti.net.ZitiSocketFactory
+import org.openziti.util.Logged
+import org.openziti.util.ZitiLog
+import java.io.IOException
+import java.net.*
+import java.nio.ByteBuffer
+import java.nio.channels.AsynchronousSocketChannel
+import java.nio.channels.CompletionHandler
+import java.nio.channels.SocketChannel
+import java.nio.channels.spi.SelectorProvider
+import java.util.concurrent.TimeUnit
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+class SyncSocketChannel(provider: SelectorProvider?) : SocketChannel(provider),
```

Comment:
not sure I agree with the name
Should it implement SelectableSocketChannel?

Reply:
Agreed - I hate the name.  I could go with just SocketChannel;  do you have a preference?

I don't see a SelectableSocketChannel.   I do see a SelectableChannel, which doesn't implement the SocketChannel interface.  I must be missing something - where should I look?